### PR TITLE
Transit machine_restart to the CPU0

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

**Comment by klogg** [comment]
I do not quite understand the description

**Comment by lorc** [comment]
> I do not quite understand the description

As do I. This patch looks like work around some deeper issue. But this is not described, so it is unclear why we need to issue `machine_restart` only on CPU0.

File: xen/arch/arm/psci.c [hunk 1]
```diff
@@ -63,8 +63,9 @@ void call_psci_cpu_off(void)
 
         /* If successfull the PSCI cpu_off call doesn't return */
         arm_smccc_smc(PSCI_0_2_FN32_CPU_OFF, &res);
-        panic("PSCI cpu off failed for CPU%d err=%d\n", smp_processor_id(),
-              PSCI_RET(res));
+        if (PSCI_RET(res) != PSCI_DENIED)
```
Comment (lorc) [inline xen/arch/arm/psci.c]: Please follow Xen coding style:
`        if ( PSCI_RET(res) != PSCI_DENIED )`

**Review by lorc — COMMENTED** [review]
It will be good to mention exact specification and locations.

Hint: this is "Arm Power State Coordination Interface (DEN0022D.b)" section 5.5.2

**Comment by lorc** [comment]
With both comments fixed:

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**